network/ipfs_impl.go

```diff
@@ -323,9 +322,6 @@ func (bsnet *impl) SendMessage(
 		return err
 	}
 
-	// TODO(https://github.com/libp2p/go-libp2p-net/issues/28): Avoid this goroutine.
-	//nolint
-	go helpers.AwaitEOF(s)
 	return s.Close()
```

Comment:
I assume the extra delay here from forcing a stream write isn't going to cause us issues and will be less problematic than the extra goroutine. Do we have any tests that will confirm this though, or is it just something we'll have to observe on internal infrastructure?

Reply:
As long as we write _something_ to the stream, this close will behave exactly like it did before. We only need to flush the handshake on close when we haven't written anything else.